craft_providers/lxd/installer.py

```diff
@@ -96,11 +101,42 @@ def is_initialized(*, remote: str, lxc: LXC) -> bool:
 
 
 def is_installed() -> bool:
-    """Check if LXD is installed (and found on PATH).
+    """Check if LXD is installed.
 
     :returns: True if lxd is installed.
     """
-    return shutil.which("lxd") is not None
+    logger.debug("Checking if LXD is installed.")
+
+    # query snapd API
+    url = "http+unix://%2Frun%2Fsnapd.socket/v2/snaps/lxd"
+    try:
+        snap_info = requests_unixsocket.get(
+            url=url, params={"select": "enabled"}
+        )
+    except requests.exceptions.ConnectionError as error:
+        raise errors.ProviderError(
+            brief="Unable to connect to snapd service."
+        ) from error
+
+    try:
+        snap_info.raise_for_status()
+    except requests.exceptions.HTTPError as error:
+        logger.debug(f"Could not get snap info for LXD: {error}")
+        return False
+
+    # the LXD snap should be installed and active but check the status
+    # for completeness
+    try:
+        status = snap_info.json()["result"]["status"]
+    except KeyError:
+        raise errors.ProviderError(
+            brief="Unexpected response from snapd service."
+        )
+
+    logger.debug(f"LXD snap status: {status}")
+    # snap status can be "installed" or "active" - "installed" revisions
+    # are filtered from this API call with `select: enabled`
+    return bool(status == "active")
```

Comment:
What do you think about retaining the old check as well?  It's unlikely to fail but may still be useful:
```
return bool(status == "active) and shutil.which("lxd") is not None
```

Reply:
@dariuszd21 also raised the issue about lxd on arch and nixos not being snap-packaged, we may need a dual check to cover these cases too.

Reply:
Yikes, that's a good point.

Does that stubbed LXD exec file on Ubuntu come in as an apt package?  Or is that just part of the base image?  

I'm wondering if craft-providers could check if LXD is installed as a snap only on Ubuntu with [`is_ubuntu_like`](https://github.com/canonical/craft-platforms/blob/a062dc03403c0404b09f6bb483b1151bb36d02f9/craft_platforms/_distro.py#L147).

Making an API call to LXC may be the most robust check.

Reply:
Covered this for now by verifying the presence of a non-snap lxd socket in the standard location, but a more robust verification would be always welcome.